.github/workflows/main.yml

```diff
@@ -0,0 +1,125 @@
+on:
+  pull_request:
+  push:
+
+jobs:
+  build:
+    runs-on: Ubuntu-latest
+    strategy:
+      matrix:
+        node-version:
+        - 14.x
+
+    steps:
+    - name: Fetch the src
+      uses: actions/checkout@v2
+    - name: >-
+        Set up NodeJS ${{ matrix.node-version }}
+        with the global NPM registry
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        registry-url: https://registry.npmjs.org
+
+    - name: Build a package tarball artifact
+      run: npm pack
+
+    - name: Save the package tarball as a GHA artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: npm-package-tarball
+        path: >-
+          ansible-ansible-language-server-*.tgz
+
+  test:
+    needs:
+    - build
+
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        node-version:
+        - 16.x
+        - 14.x
+        - 12.x
+        - 10.x
+        os:
+        - Ubuntu
+        - macOS
+        - Windows
+
+    steps:
+    - name: Fetch the GHA artifact with the package tarball
+      uses: actions/download-artifact@v2
+      with:
+        name: npm-package-tarball
+    - name: Install the package
+      run: npm i -g ansible-ansible-language-server-*.tgz
+      shell: bash
+    - name: Uninstall the package
+      run: npm un -g @ansible/ansible-language-server
+      shell: bash
+
+    - name: Clean up the checkout
+      run: rm -rfv *
+      shell: bash
+
+    - name: Fetch the src snapshot
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+
+    - name: Populate the test dependencies
+      # run: npm ci
+      run: npm i
+    - name: Run testing against the Git checkout
+      run: npm test
+
+
+  publish:
```

Comment:
Remove publish from initial version, we need a separated pipeline for that, one that would use a "release" GHA environment (security issue).

Note, doing the packaging is ok in testing pipeline, but not publishing. To simplify the setup I would just run packaging after we compile and before we even run tests. If packaging fails, there is no reason to run the tests.

Reply:
That's exactly what this is doing already.